bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java

```diff
@@ -104,13 +105,70 @@ public void initialize() {
         }
         updateStatus(ThingStatus.UNKNOWN);
 
+        updateState(CHANNEL_HVAC_TARGET_TEMPERATURE,
+                new QuantityType<Temperature>(car.getHvacTargetTemperature().doubleValue(), SIUnits.CELSIUS));
+
         // Background initialization:
         ScheduledFuture<?> job = pollingJob;
         if (job == null || job.isCancelled()) {
             pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
         }
     }
 
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case RenaultBindingConstants.CHANNEL_HVAC_TARGET_TEMPERATURE:
+                if (command instanceof RefreshType && !car.isDisableHvac()) {
+                    updateState(CHANNEL_HVAC_TARGET_TEMPERATURE, new QuantityType<Temperature>(
+                            car.getHvacTargetTemperature().doubleValue(), SIUnits.CELSIUS));
+                } else if (command instanceof QuantityType) {
+                    car.setHvacTargetTemperature(((QuantityType<?>) command).doubleValue());
+                    updateState(CHANNEL_HVAC_TARGET_TEMPERATURE, new QuantityType<Temperature>(
+                            car.getHvacTargetTemperature().doubleValue(), SIUnits.CELSIUS));
+                }
+                break;
+            case RenaultBindingConstants.CHANNEL_HVAC_STATUS:
+                // We can only trigger pre-conditioning of the car.
+                if (command instanceof StringType && command.toString().equals(Car.HVAC_STATUS_ON)
+                        && !car.isDisableHvac()) {
+                    final MyRenaultHttpSession httpSession = new MyRenaultHttpSession(this.config, httpClient);
+                    try {
+                        updateState(CHANNEL_HVAC_STATUS, new StringType("PENDING"));
+                        car.resetHVACStatus();
+                        httpSession.initSesssion(car);
+                        httpSession.actionHvacOn(car.getHvacTargetTemperature());
+                        scheduler.schedule(() -> {
+                            updateHvacStatus(httpSession);
+                        }, config.updateDelay, TimeUnit.SECONDS);
+                    } catch (Exception e) {
+                        logger.warn("Error My Renault Http Session.", e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    }
+                }
+                break;
+            case RenaultBindingConstants.CHANNEL_CHARGING_MODE:
+                if (command instanceof StringType) {
+                    car.setChargeMode(command.toString());
+                    String mode = car.getChargingMode();
```

Comment:
Maybe it would have been better to check the validity of the value here and log something if the value is not good, rather than in car.setChargeMode. Then car.setChargeMode can be a usual setter.

Reply:
Refactored with enums.